### PR TITLE
Add batch size option

### PR DESCRIPTION
**Pull request by billsaysthis**
@clementallen Can you give this a thorough review? I wrote a new test and it feels right but not 100%.

### CODE REVIEW

File: src/sumoLogger.js [hunk 1]
```diff
@@ -45,7 +46,9 @@ class SumoLogger {
                 ? newConfig.returnPromise
                 : true,
             clientUrl: newConfig.clientUrl || '',
+            useIntervalOnly: newConfig.useIntervalOnly || false,
```
Comment (clementallen) [inline src/sumoLogger.js]: Worth adding this config option into the README?

File: src/sumoLogger.js [hunk 2]
```diff
@@ -64,6 +67,12 @@ class SumoLogger {
         if (newConfig.returnPromise) {
             this.config.returnPromise = newConfig.returnPromise;
         }
+        if (newConfig.useIntervalOnly) {
+            this.config.useIntervalOnly;
```
Comment (clementallen) [inline src/sumoLogger.js]: Should this be `this.config.useInterval = newConfig.useIntervalOnly` or have I missed something?

File: src/sumoLogger.js [hunk 3]
```diff
@@ -73,8 +82,25 @@ class SumoLogger {
...
+            if (ready) {
+                this.stopLogSending();
+            }
+            return ready;
+        }
+    }
+
     sendLogs() {
-        if (this.pendingLogs.length === 0) {
+        if (this.pendingLogs .length === 0) {
```
Comment (clementallen) [inline src/sumoLogger.js]: Incorrect extra space added before `.length`

File: src/sumoLogger.js (hunk 1 above)
Comment (billsaysthis) [reply src/sumoLogger.js]: done

File: src/sumoLogger.js (hunk 2 above)
Comment (billsaysthis) [reply src/sumoLogger.js]: Yes, you're correct. 😄

File: src/sumoLogger.js (hunk 3 above)
Comment (billsaysthis) [reply src/sumoLogger.js]: fixed

Comment (billsaysthis) [reply src/sumoLogger.js]: @clementallen can you review changes? Thanks!

**Pull request merged**